File: backend/app/models/schemas/interview.py
"Interview module Pydantic request/response models."
from datetime import datetime
from typing import Any
from pydantic import BaseModel, Field

class InterviewStartRequest(BaseModel):
    job_id: int | None = Field(None, description="Target job ID")
    occupation_role_id: int = Field(..., description="Occupation role ID")

class InterviewMessageRequest(BaseModel):
    session_id: int = Field(..., description="Interview session ID")
    question_id: int = Field(..., description="Question ID")
    answer: str = Field(..., description="User answer text")

class InterviewQuestionItem(BaseModel):
    question_id: int
    content: str
    question_type: str = "TECHNICAL"
    order_no: int = 1

class InterviewStartResponse(BaseModel):
    session_id: int
    status: str = "IN_PROGRESS"
    first_question: InterviewQuestionItem | None = None

class InterviewMessageResponse(BaseModel):
    next_question: InterviewQuestionItem | None = None
    overall_score: float | None = None
    is_finished: bool = False

class InterviewReportResponse(BaseModel):
    session_id: int
    overall_score: float | None = None
    radar: dict[str, float] | None = None
    feedback: dict[str, Any] | None = None
    created_at: datetime | None = None

class QuestionBankQuery(BaseModel):
    job_id: int | None = None
    role: str | None = None
    question_type: str | None = None
    page: int = Field(default=1, ge=1)
    size: int = Field(default=20, ge=1, le=50)

class QuestionBankItem(BaseModel):
    id: int
    question_type: str
    content: str
    order_no: int

File: frontend/src/types/interview.ts
/** Interview module types — mirrors backend schemas (Day08). */

export interface InterviewQuestionItem {
  question_id: number
  content: string
  question_type: 'TECHNICAL' | 'BEHAVIORAL' | 'SITUATIONAL' | 'RESUME_BASED'
  order_no: number
}

export interface InterviewStartRequest {
  occupation_role_id: number
  job_id?: number | null
}

export interface InterviewStartResponse {
  session_id: number
  status: string
  first_question: InterviewQuestionItem | null
}

export interface InterviewMessageRequest {
  session_id: number
  question_id: number
  answer: string
}

export interface InterviewMessageResponse {
  next_question: InterviewQuestionItem | null
  overall_score: number | null
  is_finished: boolean
}

export interface InterviewRadar {
  communication: number
  technical: number
  problem_solving: number
  culture_fit: number
  depth: number
}

export interface InterviewFeedback {
  strengths: string[]
  weaknesses: string[]
  suggestions: string
}

export interface InterviewReportData {
  session_id: number
  overall_score: number | null
  radar: InterviewRadar | null
  feedback: InterviewFeedback | null
  created_at: string | null
}

export interface QuestionBankItem {
  id: number
  question_type: string
  content: string
  order_no: number
}

export interface QuestionBankData {
  records: QuestionBankItem[]
  total: number
  page: number
  size: number
}

export interface OccupRole {
  id: number
  name: string
  category: string
  description: string
}

/** Locally stored session record for recent list */
export interface SessionRecord {
  session_id: number
  role_name: string
  score: number | null
  created_at: string
  is_finished: boolean
}
